R/recipe_run.R

```diff
@@ -417,13 +416,28 @@ recipe_exists_artefacts <- function(info, id) {
 }
 
 recipe_unexpected_artefacts <- function(info, id) {
+  # expected artefacts
   expected <- unlist(info$artefacts[, "filenames"], use.names = FALSE)
+  # expected resources
+  resources <- c()
+  if (!is.null(info$resources)) {
+    resources <- info$resources
+  }
+  # expected dependencies
+  dependencies <- c()
+  if (!is.null(info$depends)) {
+    dependencies <- unlist(info$depends[, "filename"], use.names = FALSE)
+  }
   # we expect to see all artefacts from the config, the source file and the yml
   # config
-  expected <- c(expected, info$script, "orderly.yml")
+  expected <- c(expected, resources, dependencies, info$script, "orderly.yml")
+
+  # this is set to recursive to ensure that artefacts created in directories
+  # are tracked
+  found <- list.files(recursive = TRUE)
+  # TODO do we need to track when a user unexpectedly creates an empty directory ?
 
-  found <- list.files()
-  # what files have we found that we not contained in expected
+  # what files have we found that were not contained in expected
   unexpected <- setdiff(found, expected)
```

Comment:
I'd like to exclude README and it's variants here too.  So perhaps do this (untested) to ignore them too

```r
unexpected <- unexpected[!grepl("^readme(\.md)$", unexpected, ignore.case = TRUE)]
```